config/stored_requests.go

```diff
@@ -103,17 +115,19 @@ func resolvedStoredRequestsConfig(cfg *Configuration) {
 	amp.HTTP.Endpoint = sr.HTTP.AmpEndpoint
 	amp.CacheEvents.Endpoint = "/storedrequests/amp"
 	amp.HTTPEvents.Endpoint = sr.HTTPEvents.AmpEndpoint
-
-	// Set data types for each section
-	cfg.StoredRequests.dataType = RequestDataType
-	cfg.StoredRequestsAMP.dataType = AMPRequestDataType
-	cfg.StoredVideo.dataType = VideoDataType
-	cfg.CategoryMapping.dataType = CategoryDataType
```

Comment:
I think I'd rather see this code here than assign `dataType` inside the `validate`. `validate` probably should just validate, not set cfg values.